staging/src/k8s.io/cli-runtime/pkg/printers/omit_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type testResourcePrinter struct {
+	expected runtime.Object
+	t        *testing.T
+}
+
+func (p *testResourcePrinter) PrintObj(o runtime.Object, w io.Writer) error {
+	require.Equal(p.t, p.expected, o)
+	return nil
```

Comment:
you could return the error here, and then you wouldn't need `t` in that object

Reply:
`t` is required by `require.Equal`. `require.Equal` is useful when we need to compare objects, if the two objects are different, `require.Equal` would fail the test and print the diff which makes debugging easier.

Reply:
I have refactored the test a bit, could you take another look?